osu.Game/Screens/Play/ReplaySettings/ReplaySettingsOverlay.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Input;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input;
+
+namespace osu.Game.Screens.Play.ReplaySettings
+{
+    public class ReplaySettingsOverlay : FillFlowContainer
+    {
+        private const int fade_duration = 100;
+
+        private bool isVisible;
+
+        private bool isAvaliable;
+        /// <summary>
+        /// Allow user to interact with this overlay.
+        /// </summary>
+        public bool IsAvaliable

```

Comment:
How about doing `ReplaySettingsOverlay.Hide()` from inside `HUDOverlay.BindHitRenderer()`? Similar to how ModDisplay is done. Would remove this property + tidy up any usages.

Would also do the same for the key bind I think, further removing toggleVisibility, keeping this a pretty basic class.